packages/base/src/tools.ts

```diff
@@ -410,12 +410,29 @@ export const getFromIndexedDB = async (key: string) => {
 
 const fetchWithProxies = async <T>(
   url: string,
+  model: IJupyterGISModel,
   parseResponse: (response: Response) => Promise<T>
 ): Promise<T | null> => {
+  let settings: any = null;
+
+  try {
+    settings = await model.getSettings();
+  } catch (e) {
+    console.warn('Failed to get settings from model. Falling back.', e);
+  }
+
+  if (!settings || typeof settings !== 'object') {
+    settings = {};
+  }
+
+  if (!settings.proxyUrl) {
+    settings.proxyUrl = 'https://corsproxy.io'; // Temporary fallback for JupyterLite
+  }
+
```

Comment:
This can be simplified, I'd also not comment about "temporary" fallback since the settings registry is an optional plugin anyway

```suggestion
const proxyUrl = settings && settings.proxyUrl ? settings.proxyUrl : 'https://corsproxy.io';
```